### ISSUE TITLE
Strange behavior of "previous"

### ISSUE THREAD

**Issue by ruv** [Bushmills/yoda#16]
A transcript is following:
```
./yoda
yoda shell
 ok
: foo previous ;
./yoda: eval: line 442: syntax error: unexpected end of file
 ok
foo
./yoda: line 825: yoda_1377: command not found
```

version: commit 33110980

**Comment by Bushmills** [comment]
Thanks, a bug indeed!
Fixed in [eefd0d7](https://github.com/Bushmills/yoda/commit/367e5d87b113140367298b6c9a707d5155cdafbf#diff-eefd0d7defd4cb5a881d5e8b5dd21b3d3ee1b0e32773d4246caa7734fe231e69)

**Comment by ruv** [comment]
Yes, it works now.
